active/ItemsSourceView-Indexof/ItemsSourceView-IndexOf.md

```diff
@@ -0,0 +1,58 @@
+# Background
+<!-- Use this section to provide background context for the new API(s) 
+in this spec. -->
+The Xaml [ItemsRepeater](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.ItemsRepeater) control displays a list of objects using a specified template (similar to a [ListView](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.ListView)). You provide the list of objects with the [ItemsRepeater.ItemsSource](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.ItemsRepeater.ItemsSource) property. The list need only be an IIterable but can be an IVector for better perf. ItemsRepeater exposes a read-only [ItemsRepeater.ItemsSourceView](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.ItemsRepeater.ItemsSourceView) property that provides a view on this list (an [ItemsSourceView](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.ItemsSourceView)), which has helpers like Count and GetAt over what might be just an IIterable.
+
+Currently the ItemsSourceView class does not expose an IndexOf function. However during development, this method was used multiple times internally on derivatives of the class. To make the usage of this existing function easier on subclasses, this spec adds an IndexOf function to the ItemsSourceView.
+
+# Description
+
+The IndexOf function returns the index of the given element in the items source the ItemsSourceView is used for.
+Note that if an element is not contained in the items source, the function will return -1.
+> Spec note: returning -1 matches the behavior of for example IndexOf on [UIElementCollection](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.UIElementCollection) and List<T>.
+# Examples
+C# example
+
+```c#
+/// ************** Behavior ************ ///
+
+var items = new List<string>() { "One", "Two" };
+var itemsSourceView = new ItemsSourceView(items);
+var secondItem = itemsSourceView.GetAt(1);
+Assert.AreEqual(1, itemsSourceView.IndexOf(secondItem);
+
+
+/// ****** Use case in event handler ****** ///
+
+public void ItemInvoked(object sender, RoutedEventArgs e)
+{
+  int index = myItemsSourceView.IndexOf(sender); 
```

Comment:
IVectorView::IndexOf follows a different pattern. (Returns boolean indicating found/not found, and writes the index to an out parameter.)

Maybe ItemsSourceView should just implement `IVectorView<IInspectable>`? Then C# projection does the work of making it behave like an `IReadOnlyList<object>`. (Then again, `IReadOnlyList<object>` doesn't have an IndexOf method or extension method.)

Reply:
My main concern is if there is any benefit to being an IVectorView<IInspectable> except adding `GetMany` and creating this association of "this is an IVectorView"? After all, currently we only implement 2/4 of this interface, and with this spec 3/4.

What would happen to the projection of IndexOf into .Net when we use the IVectorView definition? For me it seems a bit counterintuitive to use an `out int` to get the index, especially when the method is called **IndexOf** and returns a bool. Looking at the documentation of IReadOnlyList, it doesn't seem like it does provide a function for this.

Reply:
IVector projects to C# as IList, which has an IndexOf, unlike the ICollection interface it derives from.

But IVectorView projects to IReadOnly**Collection**, which was invented for that purpose. But maybe it should have projected to an IReadOnlyList instead.

Reply:
So to the original point, we don't have one precedent we can follow; C++ precedence (IVector) is to return a bool, C# precedence (IList) is to return an int. I think we should go with the C# precedence (int), consistent with markup where we have x:null rather than x:nullptr.